lib/src/client.rs

```diff
@@ -260,14 +261,16 @@ impl ClientInner {
                 _ => true,
             }
         });
-        let mut peer_contact = PeerContact::new(
+        let peer_contact = PeerContact::new(
             peer_contact_addresses,
             identity_keypair.public(),
             provided_services,
-            None,
+            SystemTime::now()
```

Comment:
Question now is if we even want to keep the `timestamp` as one of the `PeerContact::new` parameters? `timestamp` can now get constructed inside the `new` method.

Reply:
Hmm, doesn't really hold. It needs to be flexible for the `test_housekeeping()` test.